casper/src/main/scala/coop/rchain/casper/TuranOracle.scala

```diff
@@ -28,10 +29,18 @@ import scala.collection
  * If all validators are indeed part of the clique, minMaxCliqueWeight will hopefully be equal to total_weight
  * and is_safe will reduce to total_weight >= total_weight and evaluate to true.
  */
-class TuranOracle(blocks: collection.Map[ByteString, BlockMessage],
-                  latestBlocks: collection.Map[ByteString, BlockMessage],
-                  faultToleranceThreshold: Float) {
-  def isSafe(estimate: BlockMessage): Boolean = {
+trait TuranOracle[F[_]] {
```

Comment:
This trait seems more general than just for Turan Oracle, same interface would work for any safety oracle. Maybe consider renaming to `SafetyOracle`

Reply:
Done